src/Corax/Indexing/IndexWriter.cs

```diff
@@ -687,28 +685,50 @@ private void ResetWriter()
             _tempListBuffer = new NativeIntegersList(_entriesAllocator);
             _termsPerEntryId = new NativeList<NativeList<RecordedTerm>>();
             _termsPerEntryIds = new NativeList<long>();
-
+            _numberOfModifications = 0;
+            _numberOfTermModifications = 0;
+            _initialNumberOfEntries = _indexMetadata?.ReadInt64(Constants.IndexWriter.NumberOfEntriesSlice) ?? 0;
             _pforDecoder = new FastPForDecoder(_entriesAllocator);
         }
 
-        public void TryDeleteEntryByField(string field, string term)
+        public void TryDeleteEntryByField(string field, string term, long? numberOfModifications = null)
         {
-            using var __ = Slice.From(_entriesAllocator, term, ByteStringType.Immutable, out var termSlice);
-            using var ___ = Slice.From(_entriesAllocator, field, ByteStringType.Immutable, out var fieldSlice);
-            if (TryGetEntryTermId(fieldSlice, termSlice.AsSpan(), out long idInTree) == false) 
+            if (_fieldsMapping.TryGetByFieldName(field, out var binding) && binding.FieldId == 0)
```

Comment:
Why `0` is special? Can we have a const with meaningful name for that?

Reply:
It's the primary key field id. Introduced a constant value for it.

Reply:
Would make sense to have a `PrimaryIdFieldId` constant.

Reply:
Yes, we had 0 in MappingBuilder helper anyway. We can formalize it.